src/alerts_aggregator.py

```diff
@@ -0,0 +1,242 @@
+            current_time = (datetime.combine(datetime.today(), current_time) + timedelta(minutes=15)).time()
+
+        return time_list
+    
+    # This function receives city, and range of time and returns pd series of time and its count value
+    def create_adjusted_time_column(self, settlement :str, start_time : time, end_time :time):
+        try:    
+            settlement_list = self.create_user_settlement_list(settlement)
+            is_settlement_exist = self.is_real_settlement(settlement)
+            # Extract the hour
+            start_hour = start_time.hour
+            end_hour = end_time.hour
+
+            if settlement_list != []:
+                filtered_df = self.df[self.df['data'].isin(settlement_list)]
+                filtered_df['time'] = pd.to_datetime(filtered_df['time']) 
+                # Filter the dataframe by the start and end time of the user
+                filtered_df = filtered_df[(filtered_df['time'].dt.hour >= start_hour) & (filtered_df['time'].dt.hour < end_hour)]
+                
+                # Create new column which represent the minutes by the nearest 15-minute interval
+                filtered_df['adjusted_time'] = filtered_df['time'].dt.strftime('%H:%M')
+                filtered_df['adjusted_time'] = pd.to_datetime(filtered_df['adjusted_time']).dt.round('15min').dt.strftime('%H:%M')
+                logging.debug(f'This is a quick look of how the filtered df seen: {filtered_df}')
+                
+                # Create time list of 15-minute interval and use it to map the amount of alerts for every 15-minute
+                time_list = self.quarter_hour_intervals_list(start_time, end_time)
+                logging.debug(f'This is the 15-minute interval time list: {time_list}')
+                adjusted_time_counts = filtered_df['adjusted_time'].value_counts().reindex(time_list, fill_value=0)
+                return adjusted_time_counts
+            
+            elif is_settlement_exist:
+                raise NoAlarmsException(f"There were no alarms in the settlement: {settlement}")   
+            else:
+                raise InvalidSettlement("You selected a Settlement that does not exist.")
+        
+        except NoAlarmsException as ex:
+            raise NoAlarmsException(ex)   
+        except InvalidSettlement as ex:
+            raise InvalidSettlement(ex)
+        except WrongSettlementException as ex:
+            raise WrongSettlementException(ex)
+        except requests.exceptions.RequestException as ex:
+            raise requests.exceptions.RequestException(ex)
+    
+        
+
+    # This function use create_quarter_hour_column func to return the best time to shower in
+    def best_time_to_shower(self, settlement : str, start_time : time, end_time : time) -> str:
+        try:    
+            # the function create_adjusted_time_column returns pd series of time object and its count value 
```

Comment:
this function should not return a string with a message to the user. its very wierd. 
it should only return a time. 
above thaat you should add whatever message you want

Reply:
Fix it all over this module!
that module should not be aware of any user interfacing stuff.